### PR TITLE
Update signals concept page

### PR DESCRIPTION
**Pull request by svrnm**
This PR enriches the concept page on signals with some more details on them leveraging the CNCF TAG Observability Whitepaper.

I also added a few words about events (since I had a few people now saying that otel does not have events) and about profiles

### CODE REVIEW

File: content/en/docs/concepts/signals/baggage.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Baggage
 weight: 4
+description: Contextual information that is passed between spans
```
Comment (cijothomas) [inline content/en/docs/concepts/signals/baggage.md:4]: _between spans_

Not sure if Baggage is tied to Spans in anyway..

File: content/en/docs/concepts/signals/logs.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Logs
 weight: 3
+description: A timestamped text record
```
Comment (cijothomas) [inline content/en/docs/concepts/signals/logs.md:4]: May not be strictly "text" (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#type-any).

**Review by cijothomas — CHANGES_REQUESTED** [review]
Suggesting to re-consider the wording in Baggage.
https://github.com/open-telemetry/opentelemetry.io/pull/3657/files#r1420851652

File: content/en/docs/concepts/signals/baggage.md (hunk 1 above)
Comment (cartermp) [reply content/en/docs/concepts/signals/baggage.md:4]: I would recommend just saying "between signals". Even if it's mostly used with traces today.

Comment (svrnm) [reply content/en/docs/concepts/signals/baggage.md:4]:
```suggestion
description: Contextual information that is passed between signals
```

File: content/en/docs/concepts/signals/logs.md (hunk 2 above)
Comment (svrnm) [reply content/en/docs/concepts/signals/logs.md:4]: Again, this is wording copied from the concept page about logs: https://opentelemetry.io/docs/concepts/signals/logs/, I am also not 100% happy with the wording so suggestions are welcomed (`A timestamped record` seems odd)

**Comment by svrnm** [comment]
> Suggesting to re-consider the wording in Baggage. [#3657 (files)](https://github.com/open-telemetry/opentelemetry.io/pull/3657/files#r1420851652)

Can you raise an issue for that? This is not the purpose of this PR

File: content/en/docs/concepts/signals/_index.md [hunk 3]
```diff
@@ -7,7 +7,18 @@ aliases:
...
 
-In OpenTelemetry, a **[signal][]** refers to a category of telemetry. A brief
-introduction to the currently supported signals is provided in this section.
+OpenTelemetry is structured around **[signals][]**. Signals are a system's
+outputs that a human or machine can infer knowledge from. Those signals will
+vary from system to system and depend on the objective you want to accomplish.
+It can be something you want to measure at a certain point, like temperature or
+RAM usage, or an event that goes through many components of your distributed
+system that you'd like to trace down. Signals can all be used together to
+provide different ways to observe the same piece of technology.
```
Comment (theletterf) [inline content/en/docs/concepts/signals/_index.md:16]:
```suggestion
The purpose of OpenTelemetry is to collect, process, and export **[signals][]**.
Signals are system outputs that describe the underlying activity of the operating
system and applications running on a platform. A signal can be something you
want to measure at a specific point in time, like temperature or memory usage,
or an event that goes through the components of your distributed system that
you'd like to trace. You can group different signals together to observe the
inner workings of the same piece of technology under different angles.
```

A suggested rewrite, feel free to take anything from it!

**Comment by cijothomas** [comment]
> > Suggesting to re-consider the wording in Baggage. [#3657 (files)](https://github.com/open-telemetry/opentelemetry.io/pull/3657/files#r1420851652)
> 
> Can you raise an issue for that? This is not the purpose of this PR

Done - https://github.com/open-telemetry/opentelemetry.io/issues/3674

File: content/en/docs/concepts/signals/logs.md (hunk 2 above)
Comment (svrnm) [inline content/en/docs/concepts/signals/logs.md:4]:
```suggestion
```

Comment (svrnm) [reply content/en/docs/concepts/signals/logs.md]: I removed that `description` for now. This creates inconsistency but as of now I don't have a better wording.

**Pull request merged**